src/services/utm-cookies-manager.js

```diff
@@ -0,0 +1,46 @@
+import { extractParameter } from '../utils';
+import queryString from 'query-string';
+export class UtmCookiesManager {
+  constructor() {
+    this.hasRegistered = false;
+  }
+
+  getParameter(location, parameter) {
+    return extractParameter(location, queryString.parse, parameter);
+  }
+
+  getSource(location, referrer) {
```

Comment:
you could do something like this:

`getSource(location, referrer) {
  return this.getParameter(location, 'utm_source') ?? (referrer || 'direct');
}`

Reply:
Sounds good, but I'll bring all the parameters processed into this service to avoid duplication.